analysis/src/Commands.ml

```diff
@@ -176,6 +176,78 @@ let documentSymbol ~path =
     in
     print_endline ("[\n" ^ (allSymbols |> String.concat ",\n") ^ "\n]")
 
+let rename ~path ~line ~col ~newName =
+  let uri = Uri2.fromPath path in
+  let result =
+    match ProcessCmt.getFullFromCmt ~uri with
+    | None -> Protocol.null
+    | Some full -> (
+      let pos = Utils.protocolLineColToCmtLoc ~line ~col in
+      match References.locItemForPos ~full pos with
+      | None -> Protocol.null
+      | Some locItem ->
+        let allReferences = References.allReferencesForLocItem ~full locItem in
+        let referencesToToplevelModules, referencesToItems =
+          allReferences
+          |> List.fold_left
+               (fun acc (uri2, references) ->
+                 (references |> List.map (fun loc -> (uri2, loc))) @ acc)
+               []
+          |> List.partition (fun (_, loc) -> Utils.isTopLoc loc)
+        in
+        let fileRenames =
+          referencesToToplevelModules
+          |> List.map (fun (uri, _) ->
+                 let path = Uri2.toPath uri in
+                 let dir = Filename.dirname path in
+                 let ext = Filename.extension path in
+                 let sep = Filename.dir_sep in
+                 let newPath = dir ^ sep ^ newName ^ ext in
+                 let newUri = Uri2.fromPath newPath in
+                 Protocol.
+                   {
+                     kind = `rename;
```

Comment:
Nit: I think it's easier to expose a constructor function in the protocol that only takes 2 parameters: `oldUri` and `newUri` so the type of kind does not leak here.

Reply:
Actually, I would just remove `kind` and use in the printing function only.